kratos/includes/shared_pointers.h

```diff
@@ -68,18 +68,27 @@ std::ostream& operator <<(std::ostream& rOStream, const Kratos::intrusive_ptr<T>
 } // namespace Kratos
 
 
-#define KRATOS_CLASS_POINTER_DEFINITION(a) typedef Kratos::shared_ptr<a > Pointer; \
-typedef Kratos::shared_ptr<a > SharedPointer; \
-typedef Kratos::weak_ptr<a > WeakPointer; \
-typedef Kratos::unique_ptr<a > UniquePointer
+#define KRATOS_CLASS_POINTER_DEFINITION(a)                \
+typedef Kratos::shared_ptr<a > Pointer;                   \
+typedef Kratos::shared_ptr<const a > ConstPointer;        \
```

Comment:
Please change the name to `PointerToConst` to prevent it to be confused with const shared<T>